stores/planship.ts

```diff
@@ -0,0 +1,258 @@
+import { defineStore, skipHydrate } from 'pinia'
+
+import { useUserStore } from '@/stores/user'
+import { useProjectsStore } from '@/stores/projects'
+
+import { Planship } from '@planship/fetch'
+
+async function getAccessToken() {
+  return fetch('/api/planshipToken').then((response) => response.text())
+}
+
+function getApiUrl() {
+  return typeof window === 'undefined' ? useRuntimeConfig().public.serverPlanshipApiUrl : useRuntimeConfig().public.clientPlanshipApiUrl
+}
+
+function createServerApiClient() {
+  // Create Planship client using API credentials (server side only)
+  return new Planship(
+    'clicker-demo',
+    getApiUrl(),
+    useRuntimeConfig().public.planshipApiClientId ,
+    useRuntimeConfig().planshipApiClientSecret,
+    useRuntimeConfig().public.webSocketUrl,
+  )
+}
+
+function createBrowserApiClient() {
+  return new Planship('clicker-demo', getApiUrl(), getAccessToken, useRuntimeConfig().public.webSocketUrl)
+}
+
+// This class should come autogenerated by Planship (as a snippet to copy/paste?)
+export class Entitlements {
+
+  private entitlementsDict: {} = {
+    'subscription-button-clicks': 0,
+    'button-clicks-per-minute': 0,
+    'max-projects': 0,
+    'premium-button': false,
+    'analytics-panel': false,
+    'project-types': [ 'Single' ]
+  }
+
+
+  constructor(entitlementsDict?: {}) {
+    if (entitlementsDict)
+      this.entitlementsDict = entitlementsDict;
+  }
+
+  get subscriptionButtonClicks(): number {
+    return this.entitlementsDict['subscription-button-clicks']
+  }
+
+  get buttonClicksPerMinute(): number {
+    return this.entitlementsDict['button-clicks-per-minute']
+  }
+
+  get maxProjects(): number {
+    return this.entitlementsDict['max-projects']
+  }
+
+  get premiumButton(): boolean {
+    return this.entitlementsDict['premium-button']
+  }
+
+  get analyticsPanel(): boolean {
+    return this.entitlementsDict['analytics-panel']
+  }
+
+  get projectTypes(): [] {
+    return this.entitlementsDict['project-types']
+  }
+}
+
+export const usePlanshipStore = defineStore('planship', () => {
+
+    const subscriptions = ref([])
+    const entitlementsDict = ref({})
+    const plans = ref([])
+    const clickAnalytics = ref({})
+    const currentUser = ref({})
+
+
+    const userStore = useUserStore()
+    const projectsStore = useProjectsStore()
+    const { ssrContext } = useNuxtApp();
+    const apiClient = ssrContext ? createServerApiClient() : createBrowserApiClient();
+
+    const defaultSubscription = computed(() => subscriptions.value.length > 0 ? subscriptions.value[0] : undefined)
+
+    const entitlements = computed(() => new Entitlements(entitlementsDict.value))
+
+    const currentPlanSlug = computed(() => {
+      return defaultSubscription.value ? defaultSubscription.value.plan.slug : ''
```

Comment:
-> `return defaultSubscription?.value.plan.slug : ''`

This goes for any computed prop with similar logic.

Reply:
Changed a bunch of them, however, I don't believe that you can specify a default value for this optional chains like you did above? (`: ''`). I believe it just returns `undefined`?